src/core/gui/XournalView.cpp

```diff
@@ -287,6 +289,22 @@ auto XournalView::onKeyPressEvent(GdkEventKey* event) -> bool {
         return true;
     }
 
+    // Switch color on number key
+    auto colors = control->getWindow()->getToolMenuHandler()->getColorToolItems();
+    if (!colors) {
+        return false;
+    }
+    if ((event->keyval >= GDK_KEY_0)
+        && (event->keyval < GDK_KEY_0 + std::min(10, (int)colors->size()))) {
```

Comment:
nit: use an unsigned literal instead of casting, since everything here is already unsigned to begin with.
```suggestion
        && (event->keyval < GDK_KEY_0 + std::min(10U, colors->size()))) {
```